### PR TITLE
Issue: 3482083 Remove sudo command from install script

### PR DESCRIPTION
**Pull request by anasbadaha**

### CODE REVIEW

File: plugins/SLURM-Integration/README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ Installing
 	   tar -xf ufm_slurm_integration.tar.gz
 
 ### 2) Run the installation script.
-	   ./install.sh
+	   sudo ./install.sh
```
Comment (mike-dubman) [inline plugins/SLURM-Integration/README.md:28]: script should check if active user is root
```
if [ "$EUID" -ne 0 ]
  then echo "Please run as root"
  exit
fi
```

File: plugins/SLURM-Integration/install.sh [hunk 2]
```diff
@@ -26,20 +26,85 @@ INSTALLATION_FAILED='Installation failed'
 install_status=0
 WORK_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 
+# Get the current PATH
```
Comment (mike-dubman) [inline plugins/SLURM-Integration/install.sh:29]: just need to do
```export PATH=$PATH:/some/new/path1:/new/path2```

and let ```type``` search through PATH for a binary
lines 30-38, 43-72 are not needed

File: plugins/SLURM-Integration/install.sh [hunk 3]
```diff
@@ -26,20 +26,85 @@ INSTALLATION_FAILED='Installation failed'
...
+    python3_path=$(echo "$python3_path" | cut -d ' ' -f 3)
+    echo "python3 is $python3_path"
+}
+
+function get_pip3_path()
+# This function is used to get current pip3 path using type -p pip3 command.
+{
+    pip3_path=$(type -p pip3)
+    if [[ $? -ne 0 ]];then
+        echo "Failed to run (type -p pip3) command to get pip3 path, please make sure that pip3 is installed and configured appropriately, prior to running the UFM-Slurm integration."
```
Comment (mike-dubman) [inline plugins/SLURM-Integration/install.sh:91]: all errors should be smth like
``` Error: unable to find file: $file in PATH=$PATH, please update PATH variable with correct location and rerun ```

File: plugins/SLURM-Integration/install.sh [hunk 4]
```diff
@@ -26,20 +26,85 @@ INSTALLATION_FAILED='Installation failed'
...
     declare -a intg_files=("ufm_slurm_epilog.py" "ufm_slurm_prolog.py" "ufm_slurm_utils.py" "ufm_slurm.conf" "ufm-epilog.sh" "ufm-prolog.sh" "ufm_slurm_base.py")
     for file in "${intg_files[@]}"
     do
-    sudo cp -f "$file" $SLURM_DIR
+    cp -f "$file" $SLURM_DIR
     check_failure $? "Error while copying integration file: $file"
     # skipping permission change for ufm_slurm.conf file, needs to keep the default permission which is 644.
     if [ "$file" != ${UFM_SLURM_CONF} ]; then
-        sudo chmod 755 "$SLURM_DIR/$file"
+        chmod 755 "$SLURM_DIR/$file"
```
[the remark this answers is not in the document]
Comment (mike-dubman) [reply plugins/SLURM-Integration/install.sh:107]: yep
only exe files should be 755, text files should be 600

[the remark this answers is not in the document]
Comment (anasbadaha) [reply plugins/SLURM-Integration/install.sh:107]: correct, you can see that all the files in the list in line 100 are exe files.

File: plugins/SLURM-Integration/README.md (hunk 1 above)
Comment (kedeme) [inline plugins/SLURM-Integration/README.md:28]: or run the ./install.sh with root permissions

File: plugins/SLURM-Integration/install.sh [hunk 5]
```diff
@@ -26,20 +26,56 @@ INSTALLATION_FAILED='Installation failed'
...
+    fi
+}
+
+function get_python3_path()
+# This function is used to get current python3 path using type -p python3 command.
+{
+    echo "Searching through $PATH for python3 binary..."
+    python3_path=$(type -p python3)
+    if [[ $? -ne 0 ]];then
+        echo "Error: unable to find python3 in PATH=$PATH, please update PATH variable with correct location and rerun"
```
Comment (kedeme) [inline plugins/SLURM-Integration/install.sh:49]: with the valid path of Python3 and rerun

File: plugins/SLURM-Integration/install.sh [hunk 6]
```diff
@@ -26,20 +26,56 @@ INSTALLATION_FAILED='Installation failed'
...
+    echo "python3 is $python3_path"
+}
+
+function get_pip3_path()
+# This function is used to get current pip3 path using type -p pip3 command.
+{
+    echo "Searching through $PATH for pip3 binary..."
+    pip3_path=$(type -p pip3)
+    if [[ $? -ne 0 ]];then
+        echo "Error: unable to find pip3 in PATH=$PATH, please update PATH variable with correct location and rerun"
```
Comment (kedeme) [inline plugins/SLURM-Integration/install.sh:62]: same as above with pip3

File: plugins/SLURM-Integration/install.sh [hunk 7]
```diff
@@ -55,13 +91,38 @@ function update_slurm_conf()
...
+        sed -i -e "\$a $1=$2" $CONFIG_FILE
+    fi
+}
+
+function set_python3_path()
+{
+    local file_name=$1
+
+    if [[ ! -w "$SLURM_DIR/$file_name" ]]; then
+        echo "File $SLURM_DIR/$file_name is not writable or does not exist."
```
Comment (kedeme) [inline plugins/SLURM-Integration/install.sh:108]: exit 2

**Review by kedeme — CHANGES_REQUESTED** [review]

**Review by kedeme — APPROVED** [review]

File: plugins/SLURM-Integration/install.sh [hunk 8]
```diff
@@ -46,7 +46,7 @@ function get_python3_path()
     echo "Searching through $PATH for python3 binary..."
     python3_path=$(type -p python3)
     if [[ $? -ne 0 ]];then
-        echo "Error: unable to find python3 in PATH=$PATH, please update PATH variable with correct location and rerun"
+        echo "Error: unable to find python3 in PATH=$PATH, please update PATH variabe with the valid path of Python3 and rerun"
```
Comment (mike-dubman) [inline plugins/SLURM-Integration/install.sh:49]: typo: variable

File: plugins/SLURM-Integration/install.sh [hunk 9]
```diff
@@ -26,20 +26,56 @@ INSTALLATION_FAILED='Installation failed'
...
 
+# Get the current PATH
+export PATH=$PATH:/usr/bin:/usr/local/bin:/opt/local/bin
+
 errorlog() {
     echo "ERROR: $1"
 }
+function is_root_user()
+{
+    if [ ! "$(id -u)" = "0" ]; then
```
Comment (mike-dubman) [inline plugins/SLURM-Integration/install.sh:37]: ```id``` is linux binary that not always installed, better to use bash/sh built-in env variables like ```EUID``` which is always be correct

**Comment by mike-dubman** [comment]
@kedeme - not all comments addressed before merge

also - need to add CI (@lennybe ) - where installer is running in pristine container env (rhel7-8, ub for fail) and connect it to repo.
@lennybe - why it is not part of CI

**Comment by kedeme** [comment]
@mike-dubman - we will fix the comments on a separate PR, I didn't want to delay the creating of the new SLURM integration build and testing the fixes by verification.

**Comment by mike-dubman** [comment]
> @mike-dubman - we will fix the comments on a separate PR, I didn't want to delay the creating of the new SLURM integration build and testing the fixes by verification.

but w/o seeing green CI - you risk providing broken build to verification and wasting their time

should be clear gate - CI passed before passing to verification

**Pull request merged**